### PR TITLE
Move the Lock acquireing to the end of ExecuteComputation and Execute…

### PR DESCRIPTION
**Pull request by JackCaoG**
@jonb377  I move the lock acquiring to a async function, hopefully this can make the gap go away..

### CODE REVIEW

**Review by yeounoh — APPROVED** [review]
LGTM thanks

**Review by yeounoh — APPROVED** [review]
LGTM thanks

File: third_party/xla_client/pjrt_computation_client.cc [hunk 1]
```diff
@@ -585,6 +582,29 @@ PjRtComputationClient::ExecuteReplicated(
     data_handles.push_back(datas);
   }
 
+  auto mwait = std::make_shared<util::MultiWait>(1);
```
Comment (will-cromar) [inline third_party/xla_client/pjrt_computation_client.cc:585]: I'd really like to consolidate these code paths somehow. I don't think it's doable within this PR though.

**Review by will-cromar — APPROVED** [review]

File: third_party/xla_client/pjrt_computation_client.cc (hunk 1 above)
Comment (JackCaoG) [reply third_party/xla_client/pjrt_computation_client.cc:585]: the `MultiWait` is on upstream now, as long as computationClient can import pytorch code, we can kill our own version.

**Comment by alanwaketan** [comment]
Good job!

**Comment by JackCaoG** [comment]
This pr fails the TPUCI with
```

Step #4 - "run_e2e_tests": test_expand_default (__main__.TestAtenXlaTensor) ... 2023-05-25 18:08:02.219448: F external/org_tensorflow/tensorflow/compiler/xla/pjrt/pjrt_future.h:228] Check failed: IsValid() 
Step #4 - "run_e2e_tests": https://symbolize.stripped_domain/r/?trace=7f5b50e1e8eb,7f5b5114872f,7f5a0c1ebf98,7f5a0ce68c9e,7f5a0ce68dde,7f5a0cec4cae,7f5a0cec7e80,7f5b4ca86b2e&map=58cc2d14dbdd84e9c16d4cad59d358aeca1c73bc:7f5a02793000-7f5a20450ca8 
Step #4 - "run_e2e_tests": E0525 18:08:02.219646     765 process_state.cc:1074] RAW: Signal 6 raised at PC: 0x7f5b50e1e8eb while already in FailureSignalHandler!
```

this is a non-spmd code path. I need to look into this..

**Comment by JackCaoG** [comment]
I am able to repo the failure locally.. looking

**Pull request merged**